datafusion/optimizer/src/analyzer/type_coercion.rs

```diff
@@ -930,7 +930,7 @@ fn coerce_case_expression(case: Case, schema: &DFSchema) -> Result<Case> {
 /// This method presumes that the wildcard expansion is unneeded, or has already
 /// been applied.
 pub fn coerce_union_schema(inputs: &[Arc<LogicalPlan>]) -> Result<DFSchema> {
-    let base_schema = inputs[0].schema();
+    let base_schema = inputs[0].schema().as_ref().clone().strip_qualifiers();
```

Comment:
I think we can achieve the same effect by modifying the following line, but avoid cloning the schema.
https://github.com/apache/datafusion/blob/69b0648b12ce3dc6e4e31e13424ea96dbd2dc8fd/datafusion/optimizer/src/analyzer/type_coercion.rs#L1000

Change `qualifier.cloned()` to `None`.

Reply:
Oh good catch, that is nicer.